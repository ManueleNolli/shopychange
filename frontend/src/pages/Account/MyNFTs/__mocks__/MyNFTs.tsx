import { divWithoutChildrenMock } from '../../../../../mocksUtils'

const MyNFTsMock = () => divWithoutChildrenMock('MyNFTs')
export default MyNFTsMock
