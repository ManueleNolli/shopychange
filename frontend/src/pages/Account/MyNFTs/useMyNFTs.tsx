import { useState, useEffect } from 'react'
import { useNavigate } from 'react-router-dom'
import { useUserContext } from '../../../context/userContext'
import useFetching from '../../../hooks/useFetching'
import {
  handlerQueryErrorCatch,
  ownedNftsQuery,
} from '../../../services/BackendService'
import { NFT } from '../../../types/components/NFT'

export default function useMyNFTs() {
  const navigate = useNavigate()
  const { userAddress, blockchainNetworkId } = useUserContext()
  const [ownedNfts, setOwnedNfts] = useState<NFT[]>([])
  const { isFetching, setIsFetching, isError, setIsError } = useFetching()

  useEffect(() => {
    const fetchOwnedNfts = async () => {
      if (!userAddress || !blockchainNetworkId) return

      setIsFetching(true)
      const response = await handlerQueryErrorCatch(
        ownedNftsQuery(blockchainNetworkId, userAddress)
      )

      if (response.error) {
        setIsError(true)
        setIsFetching(false)
        return
      }

      const result = response.result

      const ownedNfts = result.ownedNfts as NFT[]
      setOwnedNfts(ownedNfts)
      setIsFetching(false)
    }

    fetchOwnedNfts()
  }, [userAddress])

  const onNavigateToObservedCollections = () => {
    navigate('/account/dashboard/observed-collections')
  }

  return { ownedNfts, isFetching, isError, onNavigateToObservedCollections }
}
