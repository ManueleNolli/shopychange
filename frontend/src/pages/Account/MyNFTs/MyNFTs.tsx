import React from 'react'

import { Box, Center, Text } from '@chakra-ui/react'
import NFTOrCollectionViewer from '../../../components/NFTOrCollectionViewer/NFTOrCollectionViewer'
import ShadowButton from '../../../components/ShadowButton/ShadowButton'
import { colors } from '../../../styles/Colors'
import useMyNFTs from './useMyNFTs'
import QueryContainer from '../../../utils/QueryContainer/QueryContainer'

export default function MyNFTs() {
  const { ownedNfts, isFetching, isError, onNavigateToObservedCollections } =
    useMyNFTs()

  return (
    <Box>
      <QueryContainer isLoading={isFetching} isError={isError}>
        <Center flexDir="column">
          <NFTOrCollectionViewer nfts={ownedNfts} saleButton={true} />
        </Center>
        <Center flexDir="column" mt="2vh">
          <Text mb="1vh">Do not see your NFTs?</Text>
          <ShadowButton
            label={'Manage observed Collection'}
            bgColor={`${colors.primaryChakraColor}.400`}
            textColor={'white'}
            shadowColor={colors.primaryShadowColor}
            hoverColor={`${colors.primaryChakraColor}.500`}
            focusColor={`${colors.primaryChakraColor}.500`}
            onClick={onNavigateToObservedCollections}
          />
        </Center>
      </QueryContainer>
    </Box>
  )
}
