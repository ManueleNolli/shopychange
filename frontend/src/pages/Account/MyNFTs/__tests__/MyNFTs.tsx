import React from 'react'
import { render, screen } from '@testing-library/react'
import MyNFTs from '../MyNFTs'
import useMyNFTs from '../useMyNFTs'

jest.mock('../../../../components/NFTOrCollectionViewer/NFTOrCollectionViewer')

jest.mock('../useMyNFTs', () => jest.fn())

describe('MyNFTs', () => {
  test('renders successfully with mocked data', async () => {
    ;(useMyNFTs as jest.Mock).mockReturnValue({
      ownedNfts: [],
      isFetching: false,
      isError: false,
      onNavigateToObservedCollections: jest.fn(),
    })

    const { container } = render(<MyNFTs />)

    expect(screen.getByTestId('NFTOrCollectionViewer')).toBeInTheDocument()

    //snapshot
    expect(container).toMatchSnapshot()
  })
})
