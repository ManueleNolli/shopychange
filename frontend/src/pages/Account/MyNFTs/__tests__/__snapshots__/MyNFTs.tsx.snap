// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNFTs renders successfully with mocked data 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-b93e9j"
    >
      <div
        data-testid="NFTOrCollectionViewer"
      />
    </div>
    <div
      class="css-19pekhg"
    >
      <p
        class="chakra-text css-hz1u8i"
      >
        Do not see your NFTs?
      </p>
      <button
        class="chakra-button css-ha7ah3"
        data-testid="onClick-button"
        style="white-space: initial;"
        type="button"
      >
        Manage observed Collection
      </button>
    </div>
  </div>
</div>
`;
